packages/venia-concept/src/components/MiniCart/cartOptions.js

```diff
@@ -116,48 +129,104 @@ class CartOptions extends Component {
-                    <section className={classes.options}>
-                        <Suspense fallback={fallback}>
-                            <Options
-                                onSelectionChange={handleSelectionChange}
-                                product={configItem}
-                            />
-                        </Suspense>
-                    </section>
-                    <section className={classes.quantity}>
-                        <h2 className={classes.quantityTitle}>
-                            <span>Quantity</span>
-                        </h2>
-                        <Quantity
-                            initialValue={props.cartItem.qty}
-                            onValueChange={this.setQuantity}
-                        />
-                    </section>
-                </div>
-                <div className={classes.save}>
-                    <Button onClick={this.props.closeOptionsDrawer}>
-                        <span>Cancel</span>
-                    </Button>
-                    <Button
-                        priority="high"
-                        onClick={this.handleClick}
-                        disabled={isMissingOptions}
-                    >
-                        <span>Update Cart</span>
-                    </Button>
-                </div>
-                <div className={modalClass}>
-                    <LoadingIndicator>Updating Cart</LoadingIndicator>
-                </div>
+                {({ formState, formApi }) => (
+                    <>
+                        <div className={classes.focusItem}>
+                            <span className={classes.name}>{name}</span>
+                            <span className={classes.price}>
+                                <Price
+                                    currencyCode={currencyCode}
+                                    value={price}
+                                />
+                            </span>
+                        </div>
+                        <div className={classes.form}>
+                            <section className={classes.options}>
+                                <Suspense fallback={fallback}>
+                                    <Options
+                                        onSelectionChange={
+                                            handleSelectionChange
+                                        }
+                                        product={configItem}
+                                    />
+                                </Suspense>
+                            </section>
+                            <section className={classes.quantity}>
+                                <h2 className={classes.quantityTitle}>
+                                    <span>Quantity</span>
+                                </h2>
+                                <DecrementButtonFormApi formApi={formApi} />
+                                <Text
+                                    type="number"
```

Comment:
If decimals don't need to be supported, then `<input type="text" pattern="[0-9]*">` is a better option than `<input type="number">` because the virtual keyboard on touch devices is optimized for integer input.

Reply:
@brendanfalkowski 
Have resolved the same in latest commit.